### DISCUSSION TITLE
Adjusting a script and deleting character in the node

### DISCUSSION THREAD

**Discussion by bepolymathe** [freeplane/freeplane#948] [Help] [answered]
Hi, 

I wrote a script with the idea of being able to write the name of the person in quotes in a node, then extract this name to write it as a "Person" attribute.
 
```
// @ExecutionModes({on_selected_node="/menu_bar/bepolymathe/tags"})

// Set the node hashtag attributes from the #xxxxx words found
// in this node text, details and note.
// Any existing hashtags attribute that do not refer to any text is deleted.
// Author: bepolymathe
// Licence: WTFPL2

def tags = ( node.text =~ /"(.*?)"/ ).findAll().collect{ it[1] }
if( node.details ) tags += ( node.details.text =~ /"(.*?)"/ ).findAll().collect{ it[1] }
if( node.note ) tags += ( node.note.text =~ /"(.*?)"/ ).findAll().collect{ it[1] }
node.attributes.removeAll("Person")
tags.each{ node.attributes.add( "Person", it ) }
```

My problem is that the quotes stay in the node after the creation of the attribute. How to delete the quotes at the end of the operation ?

Thank you in advance for your help. I'm learning groovy; -)

**Comment by abc16361** [comment]
I found out one way to do that is to prefix and postfix the name with a special character like `@` instead of quotes `"`.  After the attribute creation, replace all @s on node text, details and note with `""`. for example, `node.text = node.text.replaceAll("@", "")`. Or just replace all quotes with empty value if that quote is not used in other places `node.text = node.text.replaceAll('\\"', '')`,

**Comment by bepolymathe** [comment, marked as answer]
Hi @abc16361 

Thanks for reply. Yes your first proposition is the best. I put the new script here for documentation. 

```
// @ExecutionModes({on_selected_node="/menu_bar/bepolymathe/tags"})

// Set the node attribute from the words found between "@" like in @FirstName Name@
// in this node text, details and note.
// Any existing @ is deleted after attribute creation.
// Author: bepolymathe
// Licence: WTFPL2

def tags = ( node.text =~ /@(.*?)@/ ).findAll().collect{ it[1] }
if( node.details ) tags += ( node.details.text =~ /@(.*?)@/ ).findAll().collect{ it[1] }
if( node.note ) tags += ( node.note.text =~ /@(.*?)@/ ).findAll().collect{ it[1] }
node.attributes.removeAll("Person")
tags.each{ node.attributes.add( "Person", it )
node.text = node.text.replaceAll("@", "") }

```

Have a good day.
